plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -4,103 +4,44 @@
-  ## Supported fields are: "InstanceId", "ImageId" (aka AMI), "InstanceType"
-  ## If the configuration is not provided or it has an empty list, no EC2 Metadata tags are applied.
-  # ec2_metadata_tags = ["InstanceId", "ImageId", "InstanceType"]
-  ##
-  ## Add tags retrieved from the EC2 Instance Tags associated with this instance.
-  ## If this configuration is not provided, or has an empty list, no EC2 Instance Tags are applied.
-  ## If this configuration contains one entry and its value is "*", then ALL EC2 Instance Tags for the instance are applied.
-  ## Note: This plugin renames the "aws:autoscaling:groupName" EC2 Instance Tag key to be spelled "AutoScalingGroupName".
-  ## This aligns it with the AutoScaling dimension-name seen in AWS CloudWatch.
-  # ec2_instance_tag_keys = ["aws:autoscaling:groupName", "Name"]
-  ##
-  ## Retrieve ebs_volume_id for the specified devices, add ebs_volume_id as tag. The specified devices are
-  ## the values corresponding to the tag key "disk_device_tag_key" in the input metric.
-  ## If this configuration is not provided, or has an empty list, no ebs volume is applied.
-  ## If this configuration contains one entry and its value is "*", then all ebs volume for the instance are applied.
-  # ebs_device_keys = ["/dev/xvda", "/dev/nvme0n1"]
-  ##
-  ## Specify which tag to use to get the specified disk device name from input Metric
-  # disk_device_tag_key = "device"
-  ##
-  ## Amazon Credentials
-  ## Credentials are loaded in the following order
-  ## 1) Assumed credentials via STS if role_arn is specified
-  ## 2) explicit credentials from 'access_key' and 'secret_key'
-  ## 3) shared profile from 'profile'
-  ## 4) environment variables
-  ## 5) shared credentials file
-  ## 6) EC2 Instance Profile
-  # access_key = ""
-  # secret_key = ""
-  # token = ""
-  # role_arn = ""
-  # profile = ""
-  # shared_credential_file = ""
-`
-
-const (
-	ec2InstanceTagKeyASG = "aws:autoscaling:groupName"
-	cwDimensionASG       = "AutoScalingGroupName"
-	mdKeyInstanceId      = "InstanceId"
-	mdKeyImageId         = "ImageId"
-	mdKeyInstaneType     = "InstanceType"
-	ebsVolumeId          = "EBSVolumeId"
-)
-
-var (
-	defaultRefreshInterval = 180 * time.Second
-	// backoff retry for ec2 describe instances API call. Assuming the throttle limit is 20 per second. 10 mins allow 12000 API calls.
-	backoffSleepArray = []time.Duration{0, 1 * time.Minute, 1 * time.Minute, 3 * time.Minute, 3 * time.Minute, 3 * time.Minute, 10 * time.Minute}
-)
+type EC2MetadataAPI interface {
+	GetInstanceIdentityDocument() (ec2metadata.EC2InstanceIdentityDocument, error)
+}
 
-type metadataLookup struct {
+type ec2MetadataLookup struct {
 	instanceId   bool
 	imageId      bool
 	instanceType bool
 }
 
-type ec2ProviderType func(*configaws.CredentialConfig) ec2iface.EC2API
-
-type ec2Metadata interface {
-	Available() bool
-	GetInstanceIdentityDocument() (ec2metadata.EC2InstanceIdentityDocument, error)
+type ec2MetadataRespond struct {
```

Comment:
Why are we putting all of this information inside of another struct? I don't see any benefit for pulling these out into a struct.

Reply:
Its the same reason for [ec2MetadataLookup](https://github.com/aws/amazon-cloudwatch-agent/pull/480/files#diff-6638b992018408bf44cac12361171fd050ff44fcc24c28f326988f4768a6dc2dR29-R33):
* Increase consolidation since its consolidate all the EC2MetadataRespond
* Less variables to be aware of when looking at `EC2Tagger interface`

Reply:
How is it less variables to be aware of if you are introducing another variable? I don't think the reasons you presented are convincing enough for this change. Are we going to be passing around this "response" struct outside of this Tagger? If not, it just makes it more confusing and harder to maintain.

Reply:
Based on my understanding, no for both EC2MetadataLookup and EC2MetadataRespond need to be passed outside the Tagger. 
Long short story, these metadata are parsed from IMDS, so why it's more confusing and harder to maintain? This follows the definition of `type structure` (```It is used to group related data to form a single unit. ```) so I'm not sure the reason why not to change it.
```
	instanceId     string
	imageId        string // aka AMI
	instanceType   string
	region         string
```
,